Sejima/Source/MUPageControl/MUPageControl.swift

```diff
@@ -0,0 +1,347 @@
+        didSet {
+            setNeedsLayout()
+        }
+    }
+
+    /// The inactive dot’s border width.
+    @IBInspectable open dynamic var borderWidth: CGFloat = 0.0 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+
+    /// The inactive dot’s border color.
+    @IBInspectable open dynamic var borderColor: UIColor = .clear {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+
+    /// The tint color to be used for the current page indicator.
+    @IBInspectable open dynamic var currentPageIndicatorTintColor: UIColor? = .black {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+
+    /// The tint color to be used for the page indicator.
+    @IBInspectable open dynamic var pageIndicatorTintColor: UIColor? = .lightGray {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+
+    /// Describes the margin around the page control.
+    @IBInspectable open dynamic var horizontalMargin: CGFloat = 0.0 {
+        didSet {
+            setNeedsUpdateConstraints()
+        }
+    }
+
+    // MARK: - Public functions
+
+    /// Define the current page index, animated or not.
+    open func set(page: Int, animated: Bool) {
+        guard page <= numberOfPages - 1 && page >= 0 else {
+            return
+        }
+        if animated == true {
+            moveToPage = page
+        } else {
+            currentPage = page
+        }
+    }
+
+    // MARK: - Private functions
+
+    private func tintColor(at position: Int) -> UIColor {
+        if tintColors.count <= position {
+            return tintColor
+        } else {
+            return tintColors[position]
+        }
+    }
+
+    private func updateNumberOfPages(_ count: Int) {
+        inactive.forEach { $0.removeFromSuperlayer() }
+        inactive = [CAShapeLayer]()
+        inactive = (0 ..< count).map {_ in
+            let newLayer = CAShapeLayer()
+            layer.addSublayer(newLayer)
+            return newLayer
+        }
+
+        layer.addSublayer(active)
+        setNeedsLayout()
+        invalidateIntrinsicContentSize()
+    }
+
+    private func update(for progress: Int) {
+        guard let targetX = inactive.first?.frame.origin.x,
+            numberOfPages > 1,
+            progress >= 0 && progress <= numberOfPages - 1 else {
+                return
+        }
+
+        resetInactive()
+        UIView.animate(withDuration: 0.25) { [weak self] in
+            self?.active.frame.origin.x = self?.inactive[self?.currentPage ?? 0].frame.origin.x ?? targetX
+        }
+    }
+
+    private func resetInactive() {
+        let yAxis = (bounds.size.height - elementSize.height) * 0.5
+        var frame = CGRect(x: horizontalMargin, y: yAxis, width: elementSize.width, height: elementSize.height)
+
+        inactive.enumerated().forEach { index, layer in
+            layer.frame = frame
+            UIView.animate(withDuration: 0.25, animations: { [weak self] in
+                guard let strongSelf = self else {
```

Comment:
inline pour le return

Reply:
So we disable closure_spacing swiftlint opt in rule? I think we can keep it here as functions are small.